pkg/cloud/vsphere/provisioner/common/templates.go

```diff
@@ -484,34 +461,51 @@ apt-get install -y \
 
 mv /usr/bin/kubeadm.dl /usr/bin/kubeadm
 chmod a+rx /usr/bin/kubeadm
+{{- end }} {{/* end install */}}
+
+
+{{ define "configure" -}}
+PORT=443
+MACHINE={{ .Machine.ObjectMeta.Name }}
+CONTROL_PLANE_VERSION={{ .Machine.Spec.Versions.ControlPlane }}
+CLUSTER_DNS_DOMAIN={{ .Cluster.Spec.ClusterNetwork.ServiceDomain }}
+POD_CIDR={{ getSubnet .Cluster.Spec.ClusterNetwork.Pods }}
+SERVICE_CIDR={{ getSubnet .Cluster.Spec.ClusterNetwork.Services }}
+NODE_LABEL_OPTION={{ if .Machine.Spec.Labels }}--node-labels={{ labelMap .Machine.Spec.Labels }}{{ end }}
+NODE_TAINTS_OPTION={{ if .Machine.Spec.Taints }}--register-with-taints={{ taintMap .Machine.Spec.Taints }}{{ end }}
+
+# Disable swap otherwise kubelet won't run
+swapoff -a
+sed -i '/ swap / s/^/#/' /etc/fstab
 
 systemctl enable docker
 systemctl start docker
+
+` +
+	"PRIVATEIP=`ip route get 8.8.8.8 | awk '{printf \"%s\", $NF; exit}'`" + `
+echo $PRIVATEIP > /tmp/.ip
+` +
+	"PUBLICIP=`ip route get 8.8.8.8 | awk '{printf \"%s\", $NF; exit}'`" + `
+
 cat > /etc/systemd/system/kubelet.service.d/20-cloud.conf << EOF
 [Service]
-Environment="KUBELET_DNS_ARGS=--cluster-dns=${CLUSTER_DNS_SERVER} --cluster-domain=${CLUSTER_DNS_DOMAIN}"
-Environment="KUBELET_EXTRA_ARGS=--cloud-provider=vsphere --cloud-config=/etc/kubernetes/cloud-config/cloud-config.yaml ${NODE_LABEL_OPTION} ${NODE_TAINTS_OPTION}"
+Environment="KUBELET_EXTRA_ARGS=--node-ip=${PUBLICIP} --cloud-provider=vsphere --cloud-config=/etc/kubernetes/cloud-config/cloud-config.yaml ${NODE_LABEL_OPTION} ${NODE_TAINTS_OPTION}"
 EOF
 # clear the content of the /etc/default/kubelet otherwise in v 1.11.* it causes failure to use the env variable set in the 20-cloud.conf file above
 echo > /etc/default/kubelet
 systemctl daemon-reload
-systemctl restart kubelet.service
-` +
-	"PRIVATEIP=`ip route get 8.8.8.8 | awk '{printf \"%s\", $NF; exit}'`" + `
-echo $PRIVATEIP > /tmp/.ip
-` +
-	"PUBLICIP=`ip route get 8.8.8.8 | awk '{printf \"%s\", $NF; exit}'`" + `
 
 # Set up kubeadm config file to pass parameters to kubeadm init.
 cat > /etc/kubernetes/kubeadm_config.yaml <<EOF
-apiVersion: kubeadm.k8s.io/v1alpha1
+apiVersion: kubeadm.k8s.io/v1alpha2
```

Comment:
Will this prevent us from deploying kubelet 1.10?

Reply:
I tried with k8s 1.10.x and the v1alpha2 does not work.. so with this we are going to support k8s 1.11.x and k8s 1.12.x

I suppose we will need to capture the support matrix for the version of k8s we can deploy using cluster-api

Reply:
We need to document this and then bring this up for discussion in both our bi-weekly provider meeting and the main cluster-api wg meeting when we come back from the holidays.

Reply:
is there a way of not dropping support v1.10?